mlx_graphs/utils/transformations.py

```diff
@@ -424,3 +424,49 @@ def coalesce(edge_index: mx.array) -> mx.array:
     edge_index = mx.stack([col, row])
 
     return edge_index
+
+
+@validate_edge_index
+def mask_isolated_nodes(
+    edge_index: mx.array, num_nodes: int, filter_isolated: bool = True
+) -> mx.array:
+    """Returns a mask with isolated nodes set to ``False`` to filter them out if needed.
```

Comment:
@Kheims I'm not sure whether `mask_isolated_nodes` is the best name here - basically with `filter_isolated` you can change the behavior to filter the non_isolated ones. Or maybe we can rename `filter_isolated`? wdyt?

Reply:
I don't like mask_isolated_nodes, wdyt about get_isolated_nodes_indices for the function ? or (return_isolated / isolated_only ) for the parameter ?

Reply:
it's not returning indices but a boolean mask here.

Reply:
`get_isolated_nodes_mask` makes sense, but I'd rename `filter_isolated` to `complement` as it's basically returning the isolated mask of the complement when set to false